### PR TITLE
Fix adapter function always needs a bind

### PR DESCRIPTION
**Pull request by PorterNan**
Update adapter to arrow function
Remove CallAdapter from public api

# What

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Comment by alkwa-msft** [comment]
Can we link the bug/issue that this change is related to in order to build some context?

File: packages/communication-react/review/communication-react.api.md [hunk 1]
```diff
@@ -66,99 +66,6 @@ export type AreParamEqual<A extends (props: any) => JSX.Element | undefined, B e
 // @public (undocumented)
 export type AreTypeEqual<A, B> = A extends B ? (B extends A ? true : false) : false;
 
-// @public (undocumented)
-export class AzureCommunicationCallAdapter implements CallAdapter {
```
Comment (JamesBurnside) [inline packages/communication-react/review/communication-react.api.md:70]: unintentionally removed?

Comment (JamesBurnside) [inline packages/communication-react/review/communication-react.api.md:70]: nice!

**Review by JamesBurnside — APPROVED** [review]

**Review by prprabhu-ms — CHANGES_REQUESTED** [review]
I don't think we should do this.

See https://github.com/Azure/communication-ui-library/pull/449#discussion_r652949367

Arrow functions create a new function object per instance of the class and are an anti-pattern.
Is it surprising to users that they need to bind methods to the object if they're using them as callback. That is standard JavaScript behavior, right?

**Review by anjulgarg — APPROVED** [review]

**Review by alcail — APPROVED** [review]

**Comment by alkwa-msft** [comment]
@PorterNan  can we close this since we did the binding as part of prathmesh's change?

**Comment by PorterNan** [comment]
Will open another one to remove some apis from our public api files

**Pull request closed without merging**